pkl-core/src/main/java/org/pkl/core/ast/builder/AstBuilder.java

```diff
@@ -899,8 +904,16 @@ public GeneratorMemberNode visitObjectEntry(ObjectEntryContext ctx) {
 
   @Override
   public GeneratorMemberNode visitObjectSpread(ObjectSpreadContext ctx) {
+    var scope = symbolTable.getCurrentScope();
+    var visitingIterable = scope.isVisitingIterable();
+    scope.setVisitingIterable(true);
+    var expr = visitExpr(ctx.expr());
+    scope.setVisitingIterable(visitingIterable);
     return GeneratorSpreadNodeGen.create(
-        createSourceSection(ctx), visitExpr(ctx.expr()), ctx.QSPREAD() != null);
+        createSourceSection(ctx),
+        expr,
+        ctx.QSPREAD() != null,
+        symbolTable.getCurrentScope().isVisitingIterable());
```

Comment:
If this scope is not aligned with the above, something went wrong, no?
```suggestion
        visitingIterable);
```